sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.Fixed.cs

```diff
@@ -193,4 +193,19 @@ public DerivedFromSerializable()
             otherField = 5;
         }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/3339
+    public class NullCoalesceAssignment
+    {
+        private string value = null;
+        private readonly string notWritten = "Default"; // Fixed
+
+        public void DoSomething()
```

Comment:
I would use the following snippet: 
```
public class NullCoalesceAssignment
{
    private string value = null;

    public void DoSomething()
    {
        value ??= "Empty";
    }
}
```
it reproduces the problem and it makes it obvious. Having another field which is not related to the test case can be confusing.